apps/rails/db/migrate/20250409021342_create_new_document_signatures.rb

```diff
@@ -0,0 +1,38 @@
+class CreateNewDocumentSignatures < ActiveRecord::Migration[8.0]
+  def up
+    create_table :document_signatures do |t|
+      t.belongs_to :document, null: false
+      t.belongs_to :user, null: false
+      t.string :title, null: false
+      t.datetime :signed_at
+      t.timestamps null: false
+    end
+    change_column_default :document_signatures, :created_at, from: nil, to: -> { "CURRENT_TIMESTAMP" }
+
+    execute <<~SQL
+      INSERT INTO document_signatures (document_id, user_id, title, signed_at)
+      SELECT
+        documents.id,
+        company_administrators.user_id,
+        'Company Representative',
+        documents.completed_at
```

Comment:
```suggestion
        CASE WHEN documents.administrator_signature THEN (documents.completed_at OR CURRENT_TIMESTAMP) ELSE NULL 
```
wdyt (didn't test the syntax but you get the idea :slightly_smiling_face:)?